src/librustc_mir/borrow_check/nll/constraint_generation.rs

```diff
@@ -33,25 +36,31 @@ pub(super) fn generate_constraints<'a, 'gcx, 'tcx>(
     mir: &Mir<'tcx>,
     mir_source: MirSource,
     liveness: &LivenessResults,
+    flow_inits: &FlowInProgress<MaybeInitializedLvals<'a, 'gcx, 'tcx>>,
+    move_data: &MoveData<'tcx>,
 ) {
     ConstraintGeneration {
         infcx,
         regioncx,
         mir,
         liveness,
         mir_source,
+        flow_inits,
+        move_data,
     }.add_constraints();
 }
 
-struct ConstraintGeneration<'cx, 'gcx: 'tcx, 'tcx: 'cx> {
-    infcx: &'cx InferCtxt<'cx, 'gcx, 'tcx>,
-    regioncx: &'cx mut RegionInferenceContext,
-    mir: &'cx Mir<'tcx>,
-    liveness: &'cx LivenessResults,
+struct ConstraintGeneration<'a, 'cx: 'a, 'gcx: 'tcx, 'tcx: 'cx> {
```

Comment:
Rather than this, let's rename `'a` to `'cx` and rename `'cx` to `'init`. My reasoning:

- I prefer not to use `'a` or other single-letter names, which I want to reserve for things local to a function signature.
    - I know most of the compiler doesn't follow this naming scheme. =) Gotta start somewhere though!
- I prefer for `'cx` to be contravariant by convention.
    - I know the name stinks but it's the one I've been using, we can rename later.
- I prefer then to use `'init` for the lifetime associated with the `MaybeInitializedLvals` value.